python/ray/rllib/es/es.py

```diff
@@ -192,154 +168,154 @@ def _init(self):
-                self.ob_stat.increment(
-                    result.ob_sum, result.ob_sumsq, result.ob_count)
-                ob_count_this_batch += result.ob_count
+            all_eval_returns += result.eval_returns
+            all_eval_lengths += result.eval_lengths
+
+            all_noise_indices += result.noise_indices
+            all_training_returns += result.noisy_returns
+            all_training_lengths += result.noisy_lengths
+
+        assert len(all_eval_returns) == len(all_eval_lengths)
+        assert (len(all_noise_indices) == len(all_training_returns) ==
+                len(all_training_lengths))
+
+        self.episodes_so_far += num_episodes
+        self.timesteps_so_far += num_timesteps
 
         # Assemble the results.
-        noise_inds_n = np.concatenate(
-            [r.noise_inds_n for r in curr_task_results])
-        returns_n2 = np.concatenate([r.returns_n2 for r in curr_task_results])
-        lengths_n2 = np.concatenate([r.lengths_n2 for r in curr_task_results])
-        assert (noise_inds_n.shape[0] == returns_n2.shape[0] ==
-                lengths_n2.shape[0])
+        noise_indices = np.array(all_noise_indices)
+        noisy_returns = np.array(all_training_returns)
+        noisy_lengths = np.array(all_training_lengths)
+
         # Process the returns.
         if config["return_proc_mode"] == "centered_rank":
-            proc_returns_n2 = utils.compute_centered_ranks(returns_n2)
+            proc_noisy_returns = utils.compute_centered_ranks(noisy_returns)
         else:
             raise NotImplementedError(config["return_proc_mode"])
 
         # Compute and take a step.
         g, count = utils.batched_weighted_sum(
-            proc_returns_n2[:, 0] - proc_returns_n2[:, 1],
-            (self.noise.get(idx, self.policy.num_params)
-             for idx in noise_inds_n),
+            proc_noisy_returns[:, 0] - proc_noisy_returns[:, 1],
+            (self.noise.get(index, self.policy.num_params)
+             for index in noise_indices),
             batch_size=500)
-        g /= returns_n2.size
+        g /= noisy_returns.size
         assert (
             g.shape == (self.policy.num_params,) and
             g.dtype == np.float32 and
-            count == len(noise_inds_n))
+            count == len(noise_indices))
         update_ratio = self.optimizer.update(-g + config["l2coeff"] * theta)
 
-        # Update ob stat (we're never running the policy in the master, but we
-        # might be snapshotting the policy).
-        if self.policy.needs_ob_stat:
-            self.policy.set_ob_stat(self.ob_stat.mean, self.ob_stat.std)
-
         step_tend = time.time()
-        tlogger.record_tabular("EpRewMean", returns_n2.mean())
-        tlogger.record_tabular("EpRewStd", returns_n2.std())
-        tlogger.record_tabular("EpLenMean", lengths_n2.mean())
+        tlogger.record_tabular("EvalEpRewMean",
+                               np.nan if len(all_eval_returns) == 0
```

Comment:
the only difference between `np.mean([])` and this is is that the warning is suppressed right? Also, empty array evaluates to `False` so if you want to avoid the warning, a less verbose alternative would be can be
`all_eval_returns.mean() if all_eval_returns else np.nan`

Reply:
at some point this tabular logger should be shared with the rest of the algorithms ;)

Reply:
K, I think it's ok if the warning is printed. I'll fix this.